Result/ResultProtocol.swift

```diff
@@ -1,30 +1,20 @@
 //  Copyright (c) 2015 Rob Rix. All rights reserved.
 
-#if swift(>=3.0)
-	public typealias ResultErrorType = ErrorProtocol
-#else
-	public typealias ResultErrorType = ErrorType
-#endif
-
 /// A type that can represent either failure with an error or success with a result value.
-public protocol ResultType {
+public protocol ResultProtocol {
```

Comment:
I think `Type` was a much better suffix, the fact that it's a protocols is already encoded in the type.